src/GovUk.Education.ExploreEducationStatistics.Content.Services/Extensions/OrderedQueryableExtensions.cs

```diff
@@ -0,0 +1,19 @@
+#nullable enable
+using GovUk.Education.ExploreEducationStatistics.Common.Model;
+using GovUk.Education.ExploreEducationStatistics.Content.Model;
+using System.Linq;
+
+namespace GovUk.Education.ExploreEducationStatistics.Content.Services.Extensions;
+
+public static class OrderedQueryableExtensions
+{
+    public static IOrderedQueryable<FreeTextValueResult<Publication>>? ThenByReleaseType(this IOrderedQueryable<FreeTextValueResult<Publication>>? orderedQueryable)
+    {
+        return orderedQueryable?.ThenBy(publication =>
+            publication.Value.LatestPublishedReleaseVersion!.Type == ReleaseType.AccreditedOfficialStatistics ? 0 :
+            publication.Value.LatestPublishedReleaseVersion!.Type == ReleaseType.OfficialStatistics ? 1 :
+            publication.Value.LatestPublishedReleaseVersion!.Type == ReleaseType.OfficialStatisticsInDevelopment ? 2 :
+            publication.Value.LatestPublishedReleaseVersion!.Type == ReleaseType.ExperimentalStatistics ? 3 :
+            publication.Value.LatestPublishedReleaseVersion!.Type == ReleaseType.AdHocStatistics ? 4 : 5);
```

Comment:
I don't think an ordering function is the right place to be validating data. If you compare it to built-in functions, a null ordering key essentially does nothing.

If no match is found here, it defaults to the lowest rank of `5`, which I think is more useful behaviour in this case.

Reply:
I'm don't feel super strongly about it so happy to leave it, but I still think it would be worthwhile. The purpose is if someone adds a new release type, it would be easy to forget this code and so it would not order the new release type correctly. By blowing up by throwing an exception, the moment this is manually tested that will be found and the code will get fixed.

But realistically, all this prevents is the wrong order for a new release type, which isn't the end of the world and it would take 5 seconds to fix once it was noticed.